src/js/containers/PresetButtons.js

```diff
@@ -0,0 +1,99 @@
+import { connect } from 'react-redux'
+import OnScreenPresetButtonsBody from '../Templates/Shared/PresetButtonsBody'
+import uiController from '../Controllers/UIController'
+
+const mapStateToProps = (state) => {
+    var activeApp = state.activeApp;
+    var appUI = {};
+    var presetStrings = [];
+    var presets = [];
+    if (activeApp) {
+        appUI = state.ui[activeApp];
+        presetStrings = appUI.customPresets;
+    }
+
+    for (var i=0; i<presetStrings.length; i++) {
+        presets.push({
+            label: presetStrings[i],
+            name: "PRESET_" + i.toString(),
+        })
+    }
+
+    //Assign color scheme to props
+    var theme = state.theme
+    var colorScheme = null;
+    if (theme === true) { //Dark theme
+        if(appUI.nightColorScheme) {
+            colorScheme = {}
+            if(appUI.nightColorScheme.primaryColor) {
+                colorScheme["primary"] = appUI.nightColorScheme.primaryColor
+            }
+            if(appUI.nightColorScheme.secondaryColor) {
+                colorScheme["secondary"] = appUI.nightColorScheme.secondaryColor
+            }
+        }
+    } else {
+        if(appUI.dayColorScheme) { //Light theme
+            colorScheme = {}
+            if(appUI.dayColorScheme.primaryColor) {
+                colorScheme["primary"] = appUI.dayColorScheme.primaryColor
+            }
+            if(appUI.dayColorScheme.secondaryColor) {
+                colorScheme["secondary"] = appUI.dayColorScheme.secondaryColor
+            }
+        }
+    }
+
+    return {
+        presets: presets,
+        appID: activeApp,
+        theme: state.theme,
+        colorScheme: colorScheme
+    }
+}
+
+var presetsMap = {};
```

Comment:
```suggestion
var presetsTimeoutMap = {};
```
probably should specify this relates to timeouts, the current name is a bit confusing